Explorer/Assets/DCL/AvatarRendering/Wearables/ECSThumbnailProvider.cs

```diff
@@ -0,0 +1,40 @@
+using Arch.Core;
+using Cysharp.Threading.Tasks;
+using DCL.AvatarRendering.Wearables.Components;
+using DCL.AvatarRendering.Wearables.Helpers;
+using ECS;
+using ECS.Prioritization.Components;
+using System.Threading;
+using UnityEngine;
+using ThumbnailPromise = ECS.StreamableLoading.Common.AssetPromise<UnityEngine.Texture2D, ECS.StreamableLoading.Textures.GetTextureIntention>;
+
+namespace DCL.AvatarRendering.Wearables
+{
+    public class ECSThumbnailProvider : IThumbnailProvider
+    {
+        private readonly IRealmData realmData;
+        private readonly World world;
+
+        public ECSThumbnailProvider(IRealmData realmData,
+            World world)
+        {
+            this.realmData = realmData;
+            this.world = world;
+        }
+
+        public async UniTask<Sprite?> GetAsync(IAvatarAttachment avatarAttachment, CancellationToken ct)
+        {
+            if (avatarAttachment.ThumbnailAssetResult != null)
+                return avatarAttachment.ThumbnailAssetResult.Value.Asset;
+
+            WearableComponentsUtils.CreateWearableThumbnailPromise(realmData, avatarAttachment, world, PartitionComponent.TOP_PRIORITY);
+
+            // We dont create an async task from the promise since it needs to be consumed at the proper system, not here
+            // The promise's result will eventually get replicated into the avatar attachment
+            do await UniTask.Delay(250, cancellationToken: ct);
```

Comment:
Can we replace this magic number with a named constant? I am pretty sure @davidejensen introduced something similar in his Controllers. Can we unify this value?

Reply:
Yep, now the wearables & emotes grid also request thumbnails through the interface. Looks cleaner